checker/check_runner.go

```diff
@@ -126,3 +126,23 @@ func MultiCheckAnd(fns ...CheckFn) CheckFn {
 		return minResult
 	}
 }
+
+// MultiCheckResultAnd means all check results must succeed. This returns a conservative result
+// where the worst result is returned.
+func MultiCheckResultAnd(res ...CheckResult) CheckResult {
```

Comment:
Could move this to `checker/check_result.go` where there are already a bunch of fns to make CheckResult. 

Nit: rename to `MakeAndResult`?

Reply:
Thanks! added in a separate PR since I noticed a bug in testing this one.
https://github.com/ossf/scorecard/pull/571